### DISCUSSION TITLE
The validations HasMinLen() and HasMaxLen() was deprecated?

### DISCUSSION THREAD

**Discussion by Vbaungartem** [andrebaltieri/flunt#117] [General]
Recentemente tenho realizado o curso de Domínimos Ricos do Balta, e na apresentação do Curso ele utiliza as validações acima (HasMinLen() e HasMaxLen()) para a implementação de um Contract(). Porém, ao utilizar a versão atual do Flunt não encontrei as devidas validações, além disso uma das diferenças que eu notei foi que o Contract() não funciona sem passar um Type em sua declaração. 

Para resolver o problema eu fui criando Contratcs herdadas de Contract<T> customizadas  que atendiam minha necessidade porém gostaria de saber se ainda há alguma alternativa para as mudanças que encontrem ou se foram simplesmente removidas por algum motivo. 

Desde já agradeço! :D 

See in English below:

I recently started Balta's Rich Domains Course, and in the course presentation, he used the validations HasMinLen() and HasMaxLen() to implement a Contract(). However, when using the current Flunt version, I did not find the appropriate validations. Additionally, one of the differences I observed was that Contract() does not work without passing a Type in its version of the declaration.

To solve the problem, I created customized Contracts inherited from Contract<T> that met my needs. I would like to know if there are still any alternatives for the changes I found or if they were simply removed for some reason.

Best Regards!

**Comment by andrebaltieri** [comment]
@Vbaungartem a lot of things changed from version 1 to 2. If you following my course, please, use version 1.0.5.
